frontend/src/modalNewData.js

```diff
@@ -0,0 +1,203 @@
+                                                    allCheckbox.forEach(element => {
+                                                        if (event.currentTarget.checked) {
+                                                            element.checked = true
+                                                        } else {
+                                                            element.checked = false
+                                                        }
+
+                                                    });
+
+                                                }
+                                            }/>
+                                        <label className="form-check-label" htmlFor="allCheckbox">All</label>
+                                    </div>
+                                </li>
+                                {
+                                searchInputFilter.length > 1 ? filteredResults.map((data, index) => {
+                                    return(< NewDataItem key = {
+                                        `${
+                                            data.id
+                                        }`
+                                    }
+                                    data = {
+                                        data
+                                    }
+                                    index = {
+                                        index
+                                    } />)
+                                }) : (props.newData.map((data, index) => {
+                                    return(< NewDataItem key = {
+                                        `${
+                                            data.id
+                                        }`
+                                    }
+                                    data = {
+                                        data
+                                    }
+                                    index = {
+                                        index
+                                    } />)
+                                }))
+                            } </ul>
+                        </div>
+                    </div>
+                    <div className="modal-footer">
+                        <button type="button" className="btn btn-secondary" data-dismiss="modal"
+                            onClick={props.onClose}>Close</button>
+                        <button type="button" className="btn btn-primary"
+                            onClick={
+                                () => {
+                                    props.importNewData(checkboxSelected())
+                                }
+                        }>Save changes</button>
+                    </div>
+                </div>
+            </div>
+        </div>
+    </React.Fragment>
+    )
+}
+
+
+const Modal = (props : modalProp) => props.openModal ? ReactDOM.createPortal (
```

Comment:
Just a thought here, no need to change it but: it would have been nice that the `Modal` component was generic.